qemu/tests/blockdev_mirror_sync_none.py

```diff
@@ -0,0 +1,87 @@
+import time
+
+from provider.blockdev_mirror_nowait import BlockdevMirrorNowaitTest
+
+
+class BlockdevMirrorSyncModeNoneTest(BlockdevMirrorNowaitTest):
+    """Block mirror with sync mode:none"""
+
+    def _verify_file_not_exist(self, dir_list, none_existed_files):
+        session = self.clone_vm.wait_for_login()
+        try:
+            for idx, f in enumerate(none_existed_files):
+                file_path = "%s/%s" % (dir_list[idx], f)
+                cat_cmd = "ls %s" % file_path
+
+                s, o = session.cmd_status_output(cat_cmd)
+                if s == 0:
+                    self.test.fail('File (%s) exists' % f)
+                elif 'No such file' not in o.strip():
+                    self.test.fail('Unknown error: %s' % o)
+        finally:
+            session.close()
+
+    def verify_data_files(self):
+        dir_list = [self.disks_info[t][1] for t in self._source_images]
+        none_existed_files = [self.files_info[t].pop(
+            0) for t in self._source_images]
+
+        # the second file should exist
+        super(BlockdevMirrorSyncModeNoneTest, self).verify_data_files()
+
+        # the first file should not exist
+        self._verify_file_not_exist(dir_list, none_existed_files)
+
+    def generate_inc_files(self):
+        return list(map(self.generate_data_file, self._source_images))
+
+    def wait_mirror_jobs_completed(self):
+        # Sleep some time here to wait for block-mirror done, please be noted
+        # that block-mirror with sync mode "none" is quite different from
+        # others, the job status turns into 'READY' very quickly after a new
+        # file is created, and sometimes, the current-progress and
+        # total-progress are same, but in fact, the mirror is still running.
+        # This is expected.
+        time.sleep(int(self.params.get("sync_none_mirror_timeout", "20")))
+        super(BlockdevMirrorSyncModeNoneTest,
+              self).wait_mirror_jobs_completed()
+
+    def reboot_vm(self):
+        """
+        Reboot VM to make sure the data is flushed to disk, then data
+        generated after block-mirror is copied.
```

Comment:
Seems no need since this is just for the 1st dd write. Later for the 2nd one, we only use a sync command to do the flush, so it should also flush the 1st data.

Reply:
hi @ngu-niny , dd & sync commands cannot guarantee that the 1st file was flushed to disk before block-mirror operation. Refer to my test results, the 1st file was always found on the mirrored image with a mismatched md5sum, then I changed dd option (odirect->sync), which didn't work at all. So I thought the data might not be flushed completely before block-mirror, so block-mirror also copied part of data of the 1st file, finally, to make sure that the 1st file was flushed to disk before block-mirror, I added the vm reboot step, then everything works fine.